src/codegen/generators/expression_generator.rs

```diff
@@ -839,16 +853,44 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
         Ok(())
     }
 
+    fn build_parameter_struct_gep(&self, context: &CallParameterAssignment<'ink, 'b>) -> PointerValue<'ink> {
+        unsafe {
+            // Assuming we have an inheritance hierarchy of `A <- B <- C` and a call `objC(inA := 1, ...)`
+            // then in order to access `inA` in `C` we have to generate some IR of form `objC.__B.__A.inA`.
+            // Because the parent structs are always the first member of these function blocks / classes, we
+            // can simply generate a GEP with indices `0` and repeat that `depth` times followed by the
+            // actual position of the parameter. So `objC.__B.__A.inA` becomes `GEP objC, 0, 0, 0, <inA pos>`
+            let mut gep_index = vec![0; (context.depth + 1) as usize];
+            gep_index.push(context.position);
+
+            let i32_type = self.llvm.context.i32_type();
+            let gep_index = gep_index
+                .into_iter()
+                .map(|value| i32_type.const_int(value as u64, false))
+                .collect::<Vec<_>>();
+
+            self.llvm.builder.build_gep(context.parameter_struct, &gep_index, "").unwrap()
```

Comment:
Can you tell me why we aren't using `build_in_bounds_gep` here? In your PR message you mention 
> GEPing directly, i.e. not inbound

An inbound GEP is still a direct GEP, just with additional safety guarantees and it allows LLVM to optimize more aggressively. Inheritance hierarchy access is well defined and guaranteed to be within bounds of the allocated struct, or am I missing something?
I have changed this line to use `build_in_bounds_gep` and the LIT tests still pass locally